claims-related/tests/single_source_csv/single_source_csv_comp.py

```diff
@@ -0,0 +1,222 @@
+    newdf = subset[(subset.VERSION == "NEW") | (subset.VERSION == "COMMON")]
+
+    newids = columnToSet(newdf, subcol)
+    baseids = columnToSet(basedf, subcol)
+
+    added = newids - baseids
+    removed = baseids - newids
+
+    # get all the changed IDs
+    changedids = columnToSet(subset[subset.VERSION != "COMMON"], subcol)
+
+    changed = changedids - added - removed
+
+    if debug == True:
+        print("Added")
+        print(added)
+        print("Removed")
+        print(removed)
+        print("Changed")
+        print(changed)
+
+    return (subset, added | removed | changed, added, removed, changed)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description='Parse arguments')
+    parser.add_argument('--base', type=str, help='Base file with path ',default="../../data/2020_Claims_SingleSource_v1.4.csv")
+    parser.add_argument('--new', type=str, help='New file with path ',default="../../data/2021_Claims_SingleSource_v1.3.csv")
+    parser.add_argument('--measure', type=str, default=None, action= 'append',help='MeasureId to be evaluated (optional)')
+    args = parser.parse_args()
+
+    # Method to use pipenv run python <script> <Optional: Measure ID to be checked>
+    # Change these files wherenver needed
+    basefilename = args.base
+    newfilename = args.new
+
+    basefile = pd.read_csv(basefilename, dtype=str)
+    newfile = pd.read_csv(newfilename, dtype=str)
+
+    basefile.columns = basefile.columns.str.replace(' ', '_')
+    newfile.columns = newfile.columns.str.replace(' ', '_')
+    # Version to capture
+    basefile["VERSION"] = "BASE"
+    newfile["VERSION"] = "NEW"
+    # Const to capture the whole file
+    basefile["CONST"] = "CONST"
+    newfile["CONST"] = "CONST"
+
+    key = newfile.columns.values.tolist()
+    key.remove("VERSION")
+    joint = newfile.merge(basefile, on=key, how="outer", suffixes=["_new", "_base"])
+
+    pd.set_option('display.max_rows', None)
+    pd.set_option('expand_frame_repr', False)
+
+    joint["VERSION"] = joint.apply(getJoinMeta, axis=1)
+    joint = joint.drop(["VERSION_new", "VERSION_base"], axis=1)[['VERSION',
+                                                                 'Measure_ID',
+                                                                 'DATA_ELEMENT_NAME',
+                                                                 'CODING_SYSTEM',
+                                                                 'CODE',
+                                                                 'MODIFIER',
+                                                                 'PLACE_OF_SERVICE',
+                                                                 'AGE',
+                                                                 'GENDER',
+                                                                 'CONST']]
+
+    # print("Checking if join is successful ( expected output none)")
+    # joint[joint.VERSION == "NONE"]
```

Comment:
I don't know if you want to keep these commented out lines

Reply:
This is Done